plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerUtils.java

```diff
@@ -327,4 +332,26 @@ public static String getProjectAttribute(String name, String defaultValue, IServ
 	public static String getProjectAttribute(String name, String defaultValue, IProject project) {
 		return ServerUtils.getProjectAttribute(name, defaultValue, SERVER_PROJECT_QUALIFIER, project);
 	}
+	
+	public static IDeploymentConfig getDeploymentConfig(IServerAttributes attributes) {
+		String dcKey = "deploymentconfig";
+		IService service = getService(attributes);
+		if (service == null) {
+			return null;
+		}
+		//TODO use annotations instead of labels
```

Comment:
Are you able to retrieve this value from the annoations and move the constant to OpenShiftAPIAnnotations